msg_helpers_test.go

```diff
@@ -0,0 +1,108 @@
+package dns
+
+import "testing"
+
+// TestPacketDataNsec tests generated using fuzz.go and with a message pack
+// containing the following bytes: 0000\x00\x00000000\x00\x002000000\x0060000\x00\x130000000000000000000"
+// That bytes sequence created the overflow error and further permutations of that sequence were able to trigger
+// the other code paths.
```

Comment:
uhm, shouldn't those fixes part of this PR?

Reply:
Can you elaborate more ?